### ISSUE TITLE
没有N卡，核显是怎么配置的啊

### ISSUE THREAD

**Issue by cityvoice** [Akegarasu/lora-scripts#104]
我是应该装torch==2.0.0+cu118还是torch==1.13.1+cpu？装CU的话显卡不支持，装CPU的话是不是不能用GPU。另外可否在参数里添加其它参数，比如--no_cuda,--precision full, --no-half, --skip-torch-cuda-test等等，我一加就报“`train_network.py: error: unrecognized arguments: xxx`”错误。我看stable diffusion web ui启动参数里是有这些的。

**Comment by martellz** [comment]
可以自己在sd-scripts指定device

**Comment by cityvoice** [comment]
请问具体应该怎么做?是在train_network.py里修改吗，我看代码里有”            text_encoder, vae, unet, _ = train_util.load_target_model(
                args, weight_dtype, accelerator.device if args.lowram else "cpu"
            )”这么一句话是不是只要把train.ps1里$lowram参数改成0（我之前改成了1），然后把use_cpu=true加上就行了？另外，torch版本是用cpu版本还是cuda版本？我现在用的是cuda版本的

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: ***@***.***>
发送时间: 2023年4月28日 12:09
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [Akegarasu/lora-scripts] 没有N卡，核显是怎么配置的啊 (Issue #104)


可以自己在sd-scripts指定device

―

**Comment by cityvoice** [comment]
我现在可以确认我使用的是CPU了，accelerator.device打印出来是cpu，而且no_cuda也设成了true，为什么还是报“CUDA driver version is insufficient for CUDA runtime version ”这个错误，下面是我打印出来的所有参数信息

`Namespace(config_file=None, cpu=True, mps=False, multi_gpu=False, no_cuda=True, tpu=False, use_mps_device=False, dynamo_backend='no', mixed_precision='no', fp16=False, num_processes=1, num_machines=1, num_cpu_threads_per_process=8, use_deepspeed=False, use_fsdp=False, use_megatron_lm=False, gpu_ids=None, same_network=False, machine_rank=None, main_process_ip=None, main_process_port=None, rdzv_conf='', max_restarts=0, monitor_interval=5, module=False, no_python=False, main_training_function=None, downcast_bf16=False, deepspeed_config_file=None, zero_stage=None, offload_optimizer_device=None, offload_param_device=None, gradient_accumulation_steps=None, gradient_clipping=None, zero3_init_flag=None, zero3_save_16bit_model=None, deepspeed_hostfile=None, deepspeed_exclusion_filter=None, deepspeed_inclusion_filter=None, deepspeed_multinode_launcher=None, fsdp_offload_params='false', fsdp_min_num_params=100000000.0, fsdp_sharding_strategy=1, fsdp_auto_wrap_policy=None, fsdp_transformer_layer_cls_to_wrap=None, fsdp_backward_prefetch_policy=None, fsdp_state_dict_type=None, megatron_lm_tp_degree=1, megatron_lm_pp_degree=1, megatron_lm_num_micro_batches=None, megatron_lm_sequence_parallelism=None, megatron_lm_recompute_activations=None, megatron_lm_use_distributed_optimizer=None, megatron_lm_gradient_clipping=1.0, aws_access_key_id=None, aws_secret_access_key=None, skip_torch_cuda_test=True, debug=False, training_script='./sd-scripts/train_network.py', training_script_args=['--enable_bucket', '--pretrained_model_name_or_path=./sd-models/chilloutmix-Ni.safetensors', '--train_data_dir=./train/ym', '--output_dir=./output', '--logging_dir=./logs', '--log_prefix=aki', '--resolution=512,512', '--network_module=networks.lora', '--max_train_epochs=10', '--learning_rate=1e-4', '--unet_lr=1e-4', '--text_encoder_lr=1e-5', '--lr_scheduler=cosine_with_restarts', '--lr_warmup_steps=0', '--lr_scheduler_num_cycles=1', '--network_dim=32', '--network_alpha=32', '--output_name=aki', '--train_batch_size=1', '--save_every_n_epochs=2', '--mixed_precision=no', '--save_precision=float', '--seed=1337', '--cache_latents', '--prior_loss_weight=1', '--max_token_length=225', '--caption_extension=.txt', '--save_model_as=safetensors', '--min_bucket_reso=256', '--max_bucket_reso=1024', '--keep_tokens=0', '--shuffle_caption', '--lowram', '--clip_skip=2', '--optimizer_type=AdamW8bit'], func=<function launch_command at 0x0000021B67DA2440>, use_cpu=True)
`

**Comment by Akegarasu** [comment]
用不了
